File: server/src/main/java/com/example/appserver/model/MusicalInstruments.java
package com.example.appserver.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "MusicalInstruments")
public class MusicalInstruments {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private int musicalInstrumentID;

    @ManyToOne
    @JoinColumn(name = "instrumentCategoryID", nullable = false)
    private InstrumentCategories instrumentCategory;

    @ManyToOne
    @JoinColumn(name = "instrumentBrandID", nullable = false)
    private InstrumentBrands instrumentBrand;

    private String musicalInstrumentName;
    private String description;
    private String pngUrl;
    private float price;
    private int quantity;
    private boolean onSale;
}

File: android/app/src/main/java/com/example/app/model/MusicalInstruments.kt
package com.example.app.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.PrimaryKey

@Entity(foreignKeys = [
        ForeignKey(entity = InstrumentCategories::class, parentColumns = ["instrumentCategoryID"], childColumns = ["instrumentCategoryID"], onDelete = CASCADE),
        ForeignKey(entity = InstrumentBrands::class, parentColumns = ["instrumentBrandID"], childColumns = ["instrumentBrandID"], onDelete = CASCADE)
])
data class MusicalInstruments(
    @PrimaryKey(autoGenerate = true) val musicalInstrumentID: Int,
    val instrumentCategoryID: Int,
    val instrumentBrandID: Int,
    val musicalInstrumentName: String,
    val description: String,
    val pngUrl: String,
    val price: Float,
    val quantity: Int,
    val onSale: Boolean
) {
}